lib/breakfast.ex

```diff
@@ -107,16 +120,55 @@ defmodule Breakfast do
     }
   end
 
-  def fetch(params, %Field{name: name, fetcher: :default}),
+  def set_default_value(
+        %Field{name: name, type: type} = field,
+        custom_default_values,
+        cereal_default_value
+      ) do
+    default_value =
+      with :error <- Map.fetch(custom_default_values, {name, type}),
+           :error <- Map.fetch(custom_default_values, type),
+           :error <- cereal_default_value do
+        :error
+      end
+
+    %Field{field | default: default_value}
+  end
+
+  def fetch(params, field) do
+    with :error <- do_fetch(params, field), do: field.default
+  end
+
+  defp do_fetch(params, %Field{name: name, fetcher: :default}),
     do: Breakfast.Fetch.string(params, name)
 
-  def fetch(params, %Field{mod: mod, name: name, fetcher: fetcher}),
+  defp do_fetch(params, %Field{mod: mod, name: name, fetcher: fetcher}),
     do: apply_fn(mod, fetcher, [params, name])
 
+  def cast(value, %Field{caster: :default, type: {:cereal, module}}) do
+    case Breakfast.decode(module, value) do
+      %Breakfast.Yogurt{errors: [], struct: struct} ->
+        {:ok, struct}
+
+      %Breakfast.Yogurt{errors: [_ | _]} = yogurt ->
+        {:ok, yogurt}
+    end
+  end
+
   def cast(value, %Field{caster: :default, type: type}), do: Breakfast.Type.cast(type, value)
 
   def cast(value, %Field{mod: mod, caster: caster}), do: apply_fn(mod, caster, [value])
 
+  def validate(value, %Field{validator: :default, type: {:cereal, module}}) do
+    case value do
+      %^module{} ->
+        []
+
+      %Breakfast.Yogurt{errors: errors} ->
+        [errors]
+    end
+  end
```

Comment:
Validate will yank the errors out of the yogurt so they'll bubble up to the top level yogurt